routers/api/v1/repo/issue.go

```diff
@@ -439,3 +439,65 @@ func UpdateIssueDeadline(ctx *context.APIContext, form api.EditDeadlineOption) {
 
 	ctx.JSON(201, api.IssueDeadline{Deadline: &deadline})
 }
+
+// ToggleIssueStopwatch creates or stops a stopwatch for the given issue.
+func ToggleIssueStopwatch(ctx *context.APIContext) {
+	// swagger:operation POST /repos/{owner}/{repo}/issues/{index}/stopwatch/toggle issue issueToggleStopWatch
+	// ---
+	// summary: Toggle stopwatch on an issue.
+	// consumes:
+	// - application/json
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of the issue to create or stop the stopwatch on
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/empty"
+	//   "403":
+	//     description: Not repo writer or user does not have rights to toggle stopwatch
+	//   "404":
+	//     description: Issue not found
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Status(404)
+		} else {
+			ctx.Error(500, "GetIssueByIndex", err)
+		}
+
+		return
+	}
+
+	if !ctx.Repo.CanWrite(models.UnitTypeIssues) {
+		ctx.Status(403)
+		return
+	}
+
+	if !ctx.Repo.CanUseTimetracker(issue, ctx.User) {
+		ctx.Status(403)
+		return
+	}
+
+	if err := models.CreateOrStopIssueStopwatch(ctx.User, issue); err != nil {
+		ctx.ServerError("CreateOrStopIssueStopwatch", err)
+		return
+	}
+
+	ctx.Status(201)
```

Comment:
Not sure if an empty response body is really ideal here 🤔🤔

Reply:
Empty response is ok